### PR TITLE
Development environment

### PR DESCRIPTION
**Pull request by lenonleite**
We are opening a new Pull Request with documentation about installation and a new way to work with environments, following Symfony environments structures. 

=)

<a href="https://gitpod.io/#https://github.com/mautic/developer-documentation-new/pull/126"></a>

### CODE REVIEW

File: docs/development-environment/environments.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
+Environments
+############################################################
```
Comment (RCheesley) [inline docs/development-environment/environments.rst:2]: This underline is far too long - it must be the length of the title only.

```suggestion
############
```

**Review by RCheesley — CHANGES_REQUESTED** [review]
Hey @lenonleite thanks so much for pushing this PR over to the docs!

I have made some suggestions as there were quite a few issues reported with Vale and the RST syntax. 

I have provided links to the reasons why the issues are being flagged up so that you can understand what they mean!

Also, please ensure that you commit the link files when you create them with the 'make link' command!

Ping me on Slack if you need any help with any of this! Happy to run through with you how to use Vale within VSCode as it really does help you to pick things up as you write!

File: docs/development-environment/environments.rst [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+By default the structure come with 3 environments
+``.env``
+``.env.test``
+``.env.test.local``
+
+Mautic load default values filled in app/config/parameters.php file when the system is installed.
+These values can be overridden by .env structure.
+
+Development environment:
+==================
```
[the remark this answers is not in the document]
Comment (lenonleite) [reply docs/development-environment/environments.rst:34]: I didn't know, thank you, =)

File: docs/development-environment/environments.rst [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+``.env`` come with two values:
+
+.. code-block:: env
```
[the remark this answers is not in the document]
Comment (lenonleite) [reply docs/development-environment/environments.rst:37]: Yeah, I will use bash, thank you =)

**Comment by lenonleite** [comment]
@RCheesley , =)
First of all, thank you for explaining and for sending many details and links to learn. =)

And I finished the corrections based on the comments.

File: docs/development-environment/environments.rst [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+* ``.env``                contains default values for the environment variables needed by the app
+* ``.env.local``          uncommitted file with local overrides
+* ``.env.$APP_ENV``       committed environment-specific defaults
+* ``.env.$APP_ENV.local`` uncommitted environment-specific overrides
+
+Real environment variables win over ``.env`` files.
+
+.. warning:: Don't define secrets in this file, or any other committed files. Set secrets via environment variables, or through other secret management tools.
+
+.. warning:: All secrets should add via environment variables or through other secret management.
```
Comment (RCheesley) [inline docs/development-environment/environments.rst:18]:
```suggestion
```

**Review by RCheesley — CHANGES_REQUESTED** [review]
Great work on this @lenonleite - just one bit where we need to remove the duplicate warning and space beneath and then good to go! 🚀 

Note: The build failure is due to existing issues which need to be corrected in a separate PR.

File: docs/development-environment/environments.rst (hunk 4 above)
Comment (lenonleite) [reply docs/development-environment/environments.rst]: a bit later, but done.

=)

**Comment by RCheesley** [comment]
PR is good to merge - the build warnings need to be addressed in a separate PR.

**Pull request merged**